### ISSUE TITLE
Member search

### ISSUE THREAD

**Issue by ErinCall** [code-scouts/green_mercury#13]
Search for members by:
- username
- name
- member type (mentor/member)

It would be nice if this were a single input that hit each field, but it's hard to make that performant. Maybe @jasongrlicky or @elea can offer input on how it should work?

cf. #12

**Comment by jasongrlicky** [comment]
We might be able to get a lot of mileage out of just searching on username & name on that box.

Interests can tie in to the site-wide tag system when we get that, so we can instead just have a separate method for browsing tags.

Mentor/Member can be represented somehow on the user's avatar, profile, or entry in listings of users, and doesn't necessarily need to be a facet of the search.

What say you? :smile:

**Comment by ErinCall** [comment]
It seems like it's important to be able to filter by user-type. For example if you want a ruby mentor, you want a search page that lets you say "interests include 'ruby' and user_type = 'Mentor'". Right? Or maybe we want that to really be a "meet with a mentor" request anyway and this doesn't need that feature.

**Comment by ErinCall** [comment]
Also now that I think about it, are we even implementing usernames? AFAIK we're only doing real-name. If that's so, then this becomes "search by name and/or interest" which is an easier problem.

**Comment by jasongrlicky** [comment]
Yeah, Real Name (and using email address as a username) is probably the way to go!
